osf/utils/migrations.py

```diff
@@ -12,16 +14,21 @@
 
 
 @contextmanager
-def disable_auto_now_fields(model):
+def disable_auto_now_fields(model=None):
     """
     Context manager to disable updates of all auto_now fields for a given model.
+    If model=None, updates for all auto_now fields on all models will be disabled.
 
     """
+    all_app_models = list(itertools.chain(*[app.get_models() for app in apps.get_app_configs() if 'addons' in app.label or 'osf' in app.label]))
```

Comment:
Is there a reason to filter out non-osf models here?

Reply:
Not necessarily, I was just uncertain about if non-osf models should be included here. The models excluded here are:
```
 django.contrib.auth.models.Permission,
 django.contrib.auth.models.Group,
 django.contrib.contenttypes.models.ContentType,
 django.contrib.sessions.models.Session,
 django.contrib.admin.models.LogEntry,
 django_celery_beat.models.SolarSchedule,
 django_celery_beat.models.IntervalSchedule,
 django_celery_beat.models.CrontabSchedule,
 django_celery_beat.models.PeriodicTasks,
 django_celery_beat.models.PeriodicTask,
 corsheaders.models.CorsModel,
 guardian.models.UserObjectPermission,
 guardian.models.GroupObjectPermission,
```

Reply:
If those should be included, then this line could just be
```
all_models = apps.get_models()
```

Reply:
I'm just considering how we can make this code decoupled from osf code.

Additionally, it might make sense to pass a list of models rather than a single model. That way, we could always do

```python
with disable_auto_now_fields(models=get_osf_models()):
```

if we ever wanted that behavior.

Reply:
+1 to a list of models -- I thought about that while working on this ticket. 

I can change the `model` kwarg to `models`, add an `get_osf_models` helper, and the context manager can disable *all* model auto_now fields if `models` isn't specified. 

```
def get_osf_models():
    return list(itertools.chain(*[app.get_models() for app in apps.get_app_configs() if 'addons' in app.label or 'osf' in app.label]))
```
```
def disable_auto_now_fields(models=[]):
    if not models:
        models = apps.get_models()
    ... 
```
```
# most common OSF usage, probably 
with disable_auto_now_fields(models=get_osf_models()):
```
Thoughts?

Reply:
Yep, sounds good to me. Just make sure to do

```python
def disable_auto_now_fields(models=None):
```

instead of using a list as the default.